java/statestore/src/test/java/sleeper/statestore/s3/S3StateStoreIT.java

```diff
@@ -981,179 +986,64 @@ public void shouldReturnAllPartitions() throws IOException, StateStoreException
         Schema schema = Schema.builder().rowKeyFields(field).build();
-        StateStore dynamoDBStateStore = getStateStore(schema);
-        Partition rootPartition = dynamoDBStateStore.getAllPartitions().get(0);
-        Region region1 = new Region(new RangeFactory(schema).createRange(field, Long.MIN_VALUE, 1L));
-        Partition partition1 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region1)
-                .id("id1")
-                .leafPartition(true)
-                .parentPartitionId(rootPartition.getId())
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        Region region2 = new Region(new RangeFactory(schema).createRange(field, 1L, null));
-        Partition partition2 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region2)
-                .id("id2")
-                .leafPartition(true)
-                .parentPartitionId(rootPartition.getId())
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        rootPartition = rootPartition.toBuilder()
-                .leafPartition(false)
-                .childPartitionIds(Arrays.asList(partition1.getId(), partition2.getId()))
-                .build();
-        dynamoDBStateStore.atomicallyUpdatePartitionAndCreateNewOnes(rootPartition, partition1, partition2);
-        Region region3 = new Region(new RangeFactory(schema).createRange(field, 1L, 9L));
-        Partition partition3 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region3)
-                .id("id3")
-                .leafPartition(true)
-                .parentPartitionId(partition2.getId())
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        Region region4 = new Region(new RangeFactory(schema).createRange(field, 9L, null));
-        Partition partition4 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region4)
-                .id("id4")
-                .leafPartition(true)
-                .parentPartitionId(partition2.getId())
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        partition2 = partition2.toBuilder()
-                .leafPartition(false)
-                .childPartitionIds(Arrays.asList(partition3.getId(), partition4.getId()))
-                .build();
-        dynamoDBStateStore.atomicallyUpdatePartitionAndCreateNewOnes(partition2, partition3, partition4);
 
+        PartitionTree tree = new PartitionsBuilder(schema)
+                .rootFirst("root")
+                .buildTree();
+        StateStore dynamoDBStateStore = inMemoryStateStoreWithPartitions(List.of(tree.getRootPartition()));
```

Comment:
The code under test has been removed (ie. getStateStore). This is using an in memory state store instead of the S3 implementation.